analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -144,7 +144,7 @@ class GoMod(
                             type = managerName,
                             namespace = "",
                             name = projectId.name,
-                            version = projectVcs.revision
+                            version = ""
```

Comment:
I agree that using a plain revision / SHA1 seems wrong here, but while reading through https://go.dev/doc/modules/version-numbers I was wondering whether we should instead use a pseudo or pre-release version as calculated by some Go tooling (I don't know which one, but the docs indicate that a version number could be "generated").

Reply:
Looks like `go list -f '{{.Version}}'` could do the trick.

Reply:
I just tried it:
```
analyzer/src/funTest/assets/projects/synthetic/gomod$ go list -f '{{.Version}}'
template: main:1:2: executing "main" at <.Version>: can't evaluate field Version in type *load.PackagePublic
```

Reply:
My main goal was to get rid of the version as it simplifies some upcoming work.

Reply:
You might also need to pass `-m` for the module, [it seems](https://stackoverflow.com/a/69419917/1127485).

Reply:
```
analyzer/src/funTest/assets/projects/synthetic/gomod$ go list -f '{{.Version}}' -m github.com/oss-review-toolkit/ort/gomod-synthetic-test-project
frank@EPDEFRAW0010:~/devel/ort/ort/analyzer/src/funTest/assets/projects/synthetic/gomod$ echo $?
0
```

Reply:
This works:

```
$ go list -f '{{.Version}}' -m github.com/oss-review-toolkit/ort/analyzer/src/funTest/assets/projects/synthetic/gomod@main
v0.0.0-20231012190251-67ff91afd89e
```

Reply:
Anyhow, can we just make the version empty to simplify things? (As this was the original goal of this PR)

Reply:
I get that the goal was to set the project's revision to empty, but I'm questioning the rationale. You wrote:

> The only way a version can be defined for a module is via a VCS tag.

This turns out not to be true for projects, which can have a calculated pseudo-version even without tags.

So even if it's convenient for your use-case to have an empty project revision, why is it *correct*?

Reply:
Yes, as the name says its a **pseudo-version**. Only the fact that it exists does not mean we have to use it.
For other package manager we also make the version empty. To me it's unneeded complexity. 
If it's not unneeded, then we should be able to define the need. Can you help?

Reply:
> Yes, as the name says its a **pseudo**-version. Only the fact that it exists does not mean we have to use it.

And the fact that it is called "pseudo" does not mean that we should *not* use it. I believe the version is not called "pseudo" because it's bogus, but because it's calculated and in that sense "not real", but still well defined.

> For other package manager we also make the version empty. 

If I'm not mistaken, we only use an empty project version if there is absolutely no reasonable way to get a meaningful version. However, this is not the case here.

> To me it's unneeded complexity.

To me it's correctness, that unfortunately sometimes implies complexity.

> If it's not unneeded, then we should be able to tell why it is needed. Can you help?

Well, the ORT analyzer in general always had the goal to collect as many facts as possible about the analyzed projects, and we always preferred correctness over performance or complexity so far. Following that paradigm, if it possible to get a meaningful project version with reasonable effort, we should do so.

> If you really got that this was the goal to simplify, why are you proposing to make it more complex than?

Because I believe that the goal is wrong / not in line with ORT's principles.

Reply:
I'll implemented it as suggested, but after I merged my upcoming refactoring PR. Putting this on hold for now.